src/persist-client/src/internal/paths.rs

```diff
@@ -49,6 +51,60 @@ impl PartId {
     }
 }
 
+/// A component that provides information about the writer of a blob.
+/// For older blobs, this is a UUID for the specific writer instance;
+/// for newer blobs, this is a string representing the version at which the blob was written.
+/// In either case, it's used to help determine whether a blob may eventually
+/// be linked into state, or whether it's junk that we can clean up.
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone)]
+pub enum WriterKey {
+    Id(WriterId),
+    Version(String),
+}
+
+impl WriterKey {
+    /// This writes a version of the format 0-BBBB.CCCC, where BBBB is the major
+    /// version and CCCC is the minor version. (We include the 0 so we can
+    /// change versioning schemes in the future without burning another single-letter
+    /// prefix.)
+    pub fn for_version(version: &Version) -> WriterKey {
+        assert!(version.major <= 9999);
+        assert!(version.minor <= 9999);
+        WriterKey::Version(format!("0-{:04}-{:04}", version.major, version.minor))
+    }
+}
+
+impl FromStr for WriterKey {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.is_empty() {
+            return Err("empty version string".to_owned());
+        }
+
+        let key = match &s[..1] {
+            "w" => WriterKey::Id(WriterId::from_str(s)?),
+            "x" => WriterKey::Version(s[1..].to_owned()),
```

Comment:
`x`? Another option would be to keep the `w` prefix and use the rest of the string to discriminate.

Reply:
either sgtm